File: src/Entity/OuvrierCalendrier.php
<?php
namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use App\Repository\OuvrierCalendrierRepository;
use App\Entity\Utilisateur;
use App\Entity\Candidature;

#[ORM\Entity(repositoryClass: OuvrierCalendrierRepository::class)]
class OuvrierCalendrier
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: Utilisateur::class, inversedBy: 'calendrier')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Utilisateur $ouvrier = null;

    #[ORM\OneToOne(targetEntity: Candidature::class, inversedBy: 'calendar', cascade: ['persist', 'remove'])]
    #[ORM\JoinColumn(nullable: false, onDelete: "CASCADE")]
    private ?Candidature $candidature = null;

    #[ORM\Column(type: 'datetime')]
    private \DateTimeInterface $startDate;

    #[ORM\Column(type: 'datetime')]
    private \DateTimeInterface $endDate;

    #[ORM\Column(type: 'string', length: 50)]
    private string $status = 'en_attente'; // Default status is "pending"

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getOuvrier(): ?Utilisateur
    {
        return $this->ouvrier;
    }

    public function setOuvrier(?Utilisateur $ouvrier): self
    {
        $this->ouvrier = $ouvrier;
        return $this;
    }

    public function getCandidature(): ?Candidature
    {
        return $this->candidature;
    }

    public function setCandidature(?Candidature $candidature): self
    {
        $this->candidature = $candidature;
        return $this;
    }

    public function getStartDate(): \DateTimeInterface
    {
        return $this->startDate;
    }

    public function setStartDate(\DateTimeInterface $startDate): self
    {
        $this->startDate = $startDate;
        return $this;
    }

    public function getEndDate(): \DateTimeInterface
    {
        return $this->endDate;
    }

    public function setEndDate(\DateTimeInterface $endDate): self
    {
        $this->endDate = $endDate;
        return $this;
    }

    public function getStatus(): string
    {
        return $this->status;
    }

    public function setStatus(string $status): self
    {
        $this->status = $status;
        return $this;
    }
}

File: src/main/java/tn/esprit/entities/OuvrierCalendrier.java
package tn.esprit.entities;


import jakarta.persistence.*;
import javax.validation.constraints.*;   // ✅ This is what YOU should use (based on your dependencies)
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;
import java.time.LocalDateTime;

@Entity
public class OuvrierCalendrier {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    // ManyToOne: Ouvrier (User)
    @ManyToOne(optional = false)
    @JoinColumn(name = "ouvrier_id")
    private User ouvrier;

    // OneToOne: Candidature
    @OneToOne(optional = false, cascade = CascadeType.ALL)
    @JoinColumn(name = "candidature_id", nullable = false)
    private Candidature candidature;

    // Start & End Dates
    @Column(nullable = false)
    private LocalDateTime startDate;

    @Column(nullable = false)
    private LocalDateTime endDate;

    // Status with default value
    @Column(length = 50, nullable = false)
    private String status = "en_attente"; // Default value

    // Constructors
    public OuvrierCalendrier() {}

    // Getters and Setters
    public Integer getId() {
        return id;
    }

    public User getOuvrier() {
        return ouvrier;
    }

    public void setOuvrier(User ouvrier) {
        this.ouvrier = ouvrier;
    }

    public Candidature getCandidature() {
        return candidature;
    }

    public void setCandidature(Candidature candidature) {
        this.candidature = candidature;
    }

    public LocalDateTime getStartDate() {
        return startDate;
    }

    public void setStartDate(LocalDateTime startDate) {
        this.startDate = startDate;
    }

    public LocalDateTime getEndDate() {
        return endDate;
    }

    public void setEndDate(LocalDateTime endDate) {
        this.endDate = endDate;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }
}
